pages/docs/guides/priority.mdx

```diff
@@ -0,0 +1,84 @@
+export const description = 'Dynamically adjust the execution order of functions based on any data. Ideal for pushing critical work to the front of the queue.'
+
+# Priority
+
+Priority allows you to dynamically execute some runs ahead or behind others based on any data. This allows you to prioritize some jobs ahead others without the need for a separate queue. Some use cases for priority include:
+
+* Give higher priority based on a user's subscription level, for example, free vs. paid users
+* Ensure that critical work is executed before other work in the queue
+* Prioritize certain jobs for users during onboarding to give the user a better first run experience
+
+## How to configure priority
+
+<CodeGroup forceTabs>
+```ts
+export default inngest.createFunction(
+  {
+    id: "ai-generate-summary",
+    priority: {
+      // For enterprise accounts, a given function run will be prioritized
+      // ahead of functions that were enqueued up to 120 seconds ago.
+      // For all other accounts, the function will run with no priority.
+      run: "event.data.account_type == 'enterprise' ? 120 : 0",
+    },
+  },
+  { event: "ai/summary.requested" },
+  async ({ event, step }) => {
+    // This function will be prioritized based on the account type
+  }
+);
+```
+</CodeGroup>
+
+### Configuration reference
+
+* `run` - A dynamic factor [expression](/docs/guides/writing-expressions), that evaluates to seconds, to prioritize the function by. Returning a positive number will increase that priority ahead of other jobs already in the queue. Returning a negative number will delay the function run's jobs by the given value in seconds.
+
+## How priority works
+
+Functions are scheduled in a priority queue based on the time they should run. By default, all functions are enqueued at the current time (a factor of `0`). If a function has `priority` configured, Inngest evaluates the `run` expression for each new function run based on the input event's data. The `run` expression should return a factor, in seconds, (positive or negative) to adjust the priority of the function run.
+
+Expressions that return a **positive** number will **increase the priority** of the function run ahead of other jobs already in the queue by the given value in seconds. The function will be run ahead of other jobs that were enqueued up to that many seconds ago. For example, if a function run is scheduled with a factor of `120`, it will run ahead of any jobs enqueued in the last 120 seconds, given that they are still in the queue and have not completed.
+
+Expressions that return a **negative** number will **delay** the function run by the given value in seconds.
+
+**Practical example**
+
+Given we have three jobs in the queue, each which was enqueued at the following times:
+
+```plaintext
+Jobs:          [A,        B,        C       ]
```

Comment:
I like this very much, thanks

Reply:
We had something like this before, but I re-worked it a lot b/c it previously was pretty abstract, I used more clear timestamps to make it more obvious. I think this may be something that we could use a graphic for as well to make it clearer.